### PR TITLE
Internal: fix race condition when freeing TSD

### PR DESCRIPTION
**Pull request by downor**
There is a race condition (_egl_TSDMutex) introduced by recently merged
patch,

commit 36e638f3e4c70ed2a99a761be040f957d5d8028c
Author: Sun, Yi J <yi.j.sun@intel.com>
Date:   Tue Jan 23 13:54:22 2018 +0800

    Internal: fix memory leak in mesa.

This happens when it goes into _eglInitTSD then tries to acquire
_egl_TSDMutex, which is locked within the same thread because
'_egl_TSDInitialized' is set to EGL_FALSE already.
(_eglGetCurrentThread->_eglCheckedGetTSD->_eglInitTSD)

To avoid this, it is needed to postpone resetting of '_egl_TSDInitialized'
until it is finished to back up the current context.

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>

### CODE REVIEW

File: src/egl/main/eglcurrent.c [hunk 1]
```diff
@@ -72,9 +72,10 @@ static inline void _eglFiniTSD(void)
    if (_egl_TSDInitialized) {
       _EGLThreadInfo *t = _eglGetTSD();
 
-      _egl_TSDInitialized = EGL_FALSE;
       if (t && _egl_FreeTSD)
```
Comment (kalyankondapally) [inline src/egl/main/eglcurrent.c:75]: Wondering if we should just change the original patch to match something like this:
https://chromium.googlesource.com/chromiumos/overlays/chromiumos-overlay/+/master/media-libs/arc-mesa-img/files/0028-egl-automatically-call-eglReleaseThread-on-thread-te.patch

Comment (downor) [reply src/egl/main/eglcurrent.c:75]: I can take a look at this patch. But maybe Yi J may have a better idea how we can proceed? Is it possible for you to include Yi J here?

Comment (kalyankondapally) [reply src/egl/main/eglcurrent.c:75]: We did have an internal discussion and rebased to this patch now @downor

**Comment by downor** [comment]
Dropping this as the branch is rebased

**Pull request closed without merging**